### PR TITLE
Excluding microsoft.*, system.* and runtime.native.* package id's from top downloaded packages list in past 6 weeks

### PR DESCRIPTION
**Pull request by xavierdecoster**
A quick win would be to simply filter them out on the two queries used to generate the JSON blobs for these UI reports.

https://github.com/NuGet/NuGetGallery/issues/3417

### CODE REVIEW

File: src/Stats.Warehouse/Programmability/Stored Procedures/dbo.DownloadReportRecentPopularity.sql [hunk 1]
```diff
@@ -26,6 +26,9 @@ BEGIN
 			AND F.[Timestamp] <= @Cursor
 			AND C.ClientCategory NOT IN ('Crawler', 'Unknown')
 			AND NOT (C.ClientCategory = 'NuGet' AND CAST(ISNULL(C.[Major], '0') AS INT) > 10)
+			AND (P.[LowercasedPackageId] NOT LIKE 'microsoft.%'
```
Comment (joelverhagen) [inline src/Stats.Warehouse/Programmability/Stored Procedures/dbo.DownloadReportRecentPopularity.sql]: This should take the owner into account too. It should only be Microsoft-ish owners that we exclude.

**Comment by joelverhagen** [comment]
I don't think this should exclude non-"bcl" packages like `Microsoft.AspNet.Mvc` or `Microsoft.EntityFrameworkCore`. This is especially weird because `EntityFramework` (not `Core`) will not be exclude but the similar and newer product `EntityFrameworkCore` will...

**Comment by xavierdecoster** [comment]
The statistics database has no ownership information (nor package key for that matter...).

This PR was merely intended as a quick win, based on @skofman1's suggestion [here](https://github.com/NuGet/NuGetGallery/issues/3417#issuecomment-338062368), to improve things a bit (literally took me 5 mins... was out of scope for this sprint, and likely still is for quite some time).

A *proper* solution as you suggest should indeed take other factors into account, but that will take (much) more time and effort.

My vote would be, for now, to simply filter out some prefixes for now (the `WHERE` clause can be tweaked further in this PR...).

**Comment by joelverhagen** [comment]
If we really want to do a quick fix, we should do one that is more specific, like a list of specific package IDs. That's my opinion at least.

**Comment by skofman1** [comment]
The intent here is to remove microsoft packages from the most popular packages list - not only dependent packages. This will give community packages a visibility boost. 
Also, since this code runs in the statistics DB, getting owner information will not be trivial. 
We should take a look at what the list will look like after the filtering @xavierdecoster suggests here, and see if there are any other microsoft packages that we missed (like EntityFramework )

**Comment by joelverhagen** [comment]
We can do that analysis right now before committing a change. And we should have a more specific change, excluding a list of known IDs we don't care about. This approach does not exclude packages owned by the community and can cover the `EntityFramework` case.

Also, from the original issue, it's not clear that non-bcl packages like EF or MVC should even be filtered out at all.

**Comment by skofman1** [comment]
@xavierdecoster , we had an offline discussion, and I got convinced by seeing this package: https://www.nuget.org/packages/System.Data.SQLite/
The suggestion @loic-sharma and @joelverhagen have is to have a table in stats DB with excluded packages, and it will contain explicit ids. What do you think?

**Comment by joelverhagen** [comment]
Additionally, I also think we should get more input from the community (and EF + MVC team) about whether EF, MVC, WindowAzure.Storage, etc packages should be excluded. I think not since they are not the same "category" as these BCL packages. That is, users have to opt-in to using them.

**Comment by xavierdecoster** [comment]
I want to also point out that this change only affects the TOP 100 package downloads in the past 6 weeks. That is [this list](https://www.nuget.org/stats/packages), and [this one](https://www.nuget.org/stats/packageversions).

`System.Data.SQLite` does not come anywhere near that. Not sure if it would after applying the filters (did not connect to the PROD db yet to verify).

No need for an extra table, we can define a `PackageSet` (already have a table for that) with the ones to be filtered out. However, someone will need to come up with a list and check with the owners.

**Comment by loic-sharma** [comment]
`System.Data.SQLite` is just an example. We should not have code that may filter out community packages', even if the code "works" today.

**Comment by xavierdecoster** [comment]
@loic-sharma agreed, but that's the balance between a quick-win that works today (so "good enough" short term), and full-blown solution that works forever :)

**Comment by loic-sharma** [comment]
I think a blacklist would still be a quick-win that also is good enough.

**Comment by xavierdecoster** [comment]
@loic-sharma true. In either case, we need a list of package ids to blacklist from that top 100.

**Comment by loic-sharma** [comment]
I went to [this](https://www.nuget.org/stats/packages) and [this](https://www.nuget.org/stats/packageversions) list, combined the results, and removed all duplicate ids. Here are the packages I think we should filter:

```
.ubuntu.14.04-x64.runtime.native.system.net.security
microsoft.bcl
microsoft.bcl.async
microsoft.bcl.build
microsoft.codeanalysis.analyzers
microsoft.codeanalysis.common
microsoft.codeanalysis.csharp
microsoft.codeanalysis.visualbasic
microsoft.codedom.providers.dotnetcompilerplatform
microsoft.csharp
microsoft.dotnet.internalabstractions
microsoft.dotnet.platformabstractions
microsoft.net.compilers
microsoft.net.http
microsoft.net.http.headers
microsoft.net.test.sdk
microsoft.netcore.app
microsoft.netcore.dotnethost
microsoft.netcore.dotnethostpolicy
microsoft.netcore.dotnethostresolver
microsoft.netcore.jit
microsoft.netcore.platforms
microsoft.netcore.runtime.coreclr
microsoft.netcore.targets
microsoft.netcore.windows.apisets
microsoft.testplatform.objectmodel
microsoft.testplatform.testhost
microsoft.tpl.dataflow
microsoft.visualbasic
microsoft.web.infrastructure
microsoft.win32.primitives
microsoft.win32.registry
netstandard.library
runtime.any.system.collections
runtime.any.system.diagnostics.tools
runtime.any.system.diagnostics.tracing
runtime.any.system.globalization
runtime.any.system.globalization.calendars
runtime.any.system.io
runtime.any.system.reflection
runtime.any.system.reflection.extensions
runtime.any.system.reflection.primitives
runtime.any.system.resources.resourcemanager
runtime.any.system.runtime
runtime.any.system.runtime.handles
runtime.any.system.runtime.interopservices
runtime.any.system.text.encoding
runtime.any.system.text.encoding.extensions
runtime.any.system.threading.tasks
runtime.any.system.threading.timer
runtime.debian.8-x64.runtime.native.system.security.cryptography.openssl
runtime.fedora.23-x64.runtime.native.system.security.cryptography.openssl
runtime.fedora.24-x64.runtime.native.system.security.cryptography.openssl
runtime.native.system
runtime.native.system.io.compression
runtime.native.system.net.http
runtime.native.system.net.security
runtime.native.system.security.cryptography
runtime.native.system.security.cryptography.apple
runtime.native.system.security.cryptography.openssl
runtime.opensuse.13.2-x64.runtime.native.system.security.cryptography.openssl
runtime.opensuse.42.1-x64.runtime.native.system.security.cryptography.openssl
runtime.osx.10.10-x64.runtime.native.system.security.cryptography.apple
runtime.osx.10.10-x64.runtime.native.system.security.cryptography.openssl
runtime.rhel.7-x64.runtime.native.system.security.cryptography.openssl
runtime.ubuntu.14.04-x64.microsoft.netcore.dotnethost
runtime.ubuntu.14.04-x64.microsoft.netcore.dotnethostpolicy
runtime.ubuntu.14.04-x64.microsoft.netcore.dotnethostresolver
runtime.ubuntu.14.04-x64.microsoft.netcore.jit
runtime.ubuntu.14.04-x64.microsoft.netcore.runtime.coreclr
runtime.ubuntu.14.04-x64.runtime.native.system
runtime.ubuntu.14.04-x64.runtime.native.system.io.compression
runtime.ubuntu.14.04-x64.runtime.native.system.net.http
runtime.ubuntu.14.04-x64.runtime.native.system.security.cryptography
runtime.ubuntu.14.04-x64.runtime.native.system.security.cryptography.openssl
runtime.ubuntu.16.04-x64.runtime.native.system.security.cryptography.openssl
runtime.ubuntu.16.10-x64.runtime.native.system.security.cryptography.openssl
runtime.unix.microsoft.win32.primitives
runtime.unix.system.console
runtime.unix.system.diagnostics.debug
runtime.unix.system.io.filesystem
runtime.unix.system.net.primitives
runtime.unix.system.net.sockets
runtime.unix.system.private.uri
runtime.unix.system.runtime.extensions
runtime.win.microsoft.win32.primitives
runtime.win.system.console
runtime.win.system.diagnostics.debug
runtime.win.system.io.filesystem
runtime.win.system.net.primitives
runtime.win.system.net.sockets
runtime.win.system.runtime.extensions
system.appcontext
system.buffers
system.collections
system.collections.concurrent
system.collections.immutable
system.collections.nongeneric
system.collections.specialized
system.componentmodel
system.componentmodel.annotations
system.componentmodel.eventbasedasync
system.componentmodel.primitives
system.componentmodel.typeconverter
system.console
system.diagnostics.contracts
system.diagnostics.debug
system.diagnostics.diagnosticsource
system.diagnostics.fileversioninfo
system.diagnostics.process
system.diagnostics.stacktrace
system.diagnostics.textwritertracelistener
system.diagnostics.tools
system.diagnostics.tracesource
system.diagnostics.tracing
system.dynamic.runtime
system.globalization
system.globalization.calendars
system.globalization.extensions
system.io
system.io.compression
system.io.compression.zipfile
system.io.filesystem
system.io.filesystem.primitives
system.io.filesystem.watcher
system.io.memorymappedfiles
system.io.pipes
system.io.unmanagedmemorystream
system.linq
system.linq.expressions
system.linq.parallel
system.linq.queryable
system.net.http
system.net.nameresolution
system.net.primitives
system.net.requests
system.net.security
system.net.sockets
system.net.webheadercollection
system.net.websockets
system.numerics.vectors
system.objectmodel
system.private.datacontractserialization
system.private.uri
system.reflection
system.reflection.dispatchproxy
system.reflection.emit
system.reflection.emit.ilgeneration
system.reflection.emit.lightweight
system.reflection.extensions
system.reflection.metadata
system.reflection.primitives
system.reflection.typeextensions
system.resources.reader
system.resources.resourcemanager
system.resources.writer
system.runtime
system.runtime.compilerservices.unsafe
system.runtime.extensions
system.runtime.handles
system.runtime.interopservices
system.runtime.interopservices.runtimeinformation
system.runtime.loader
system.runtime.numerics
system.runtime.serialization.formatters
system.runtime.serialization.json
system.runtime.serialization.primitives
system.runtime.serialization.xml
system.security.claims
system.security.cryptography.algorithms
system.security.cryptography.cng
system.security.cryptography.csp
system.security.cryptography.encoding
system.security.cryptography.openssl
system.security.cryptography.primitives
system.security.cryptography.x509certificates
system.security.principal
system.security.principal.windows
system.spatial
system.text.encoding
system.text.encoding.codepages
system.text.encoding.extensions
system.text.encodings.web
system.text.regularexpressions
system.threading
system.threading.overlapped
system.threading.tasks
system.threading.tasks.dataflow
system.threading.tasks.extensions
system.threading.tasks.parallel
system.threading.thread
system.threading.threadpool
system.threading.timer
system.valuetuple
system.xml.readerwriter
system.xml.xdocument
system.xml.xmldocument
system.xml.xmlserializer
system.xml.xpath
system.xml.xpath.xdocument
system.xml.xpath.xmldocument
```

This filter does NOT include the following pacakges from the lists above:

```
antlr
automapper
bootstrap
castle.core
commonservicelocator
dapper
entityframework
jquery
jquery.validation
libuv
log4net
microsoft.applicationinsights
microsoft.applicationinsights.agent.intercept
microsoft.applicationinsights.dependencycollector
microsoft.applicationinsights.perfcountercollector
microsoft.applicationinsights.windowsserver
microsoft.applicationinsights.windowsserver.telemetrychannel
microsoft.aspnet.cors
microsoft.aspnet.identity.core
microsoft.aspnet.identity.entityframework
microsoft.aspnet.identity.owin
microsoft.aspnet.mvc
microsoft.aspnet.razor
microsoft.aspnet.telemetrycorrelation
microsoft.aspnet.web.optimization
microsoft.aspnet.webapi
microsoft.aspnet.webapi.client
microsoft.aspnet.webapi.core
microsoft.aspnet.webapi.cors
microsoft.aspnet.webapi.helppage
microsoft.aspnet.webapi.owin
microsoft.aspnet.webapi.webhost
microsoft.aspnet.webpages
microsoft.aspnetcore.hosting
microsoft.aspnetcore.hosting.abstractions
microsoft.aspnetcore.hosting.server.abstractions
microsoft.aspnetcore.http
microsoft.aspnetcore.http.abstractions
microsoft.aspnetcore.http.extensions
microsoft.aspnetcore.http.features
microsoft.aspnetcore.server.kestrel
microsoft.aspnetcore.webutilities
microsoft.azure.keyvault.core
microsoft.data.edm
microsoft.data.odata
microsoft.data.services.client
microsoft.extensions.configuration
microsoft.extensions.configuration.abstractions
microsoft.extensions.configuration.environmentvariables
microsoft.extensions.dependencyinjection
microsoft.extensions.dependencyinjection.abstractions
microsoft.extensions.dependencymodel
microsoft.extensions.fileproviders.abstractions
microsoft.extensions.fileproviders.physical
microsoft.extensions.filesystemglobbing
microsoft.extensions.logging
microsoft.extensions.logging.abstractions
microsoft.extensions.objectpool
microsoft.extensions.options
microsoft.extensions.platformabstractions
microsoft.extensions.primitives
microsoft.identitymodel.logging
microsoft.identitymodel.tokens
microsoft.jquery.unobtrusive.validation
microsoft.owin
microsoft.owin.host.systemweb
microsoft.owin.security
microsoft.owin.security.cookies
microsoft.owin.security.oauth
microsoft.windowsazure.configurationmanager
modernizr
moq
mstest.testadapter
mstest.testframework
newtonsoft.json
nunit
owin
respond
restsharp
serilog.sinks.file
sharpziplib
unity
webgrease
xunit
xunit.abstractions
xunit.assert
xunit.core
xunit.extensibility.core
xunit.extensibility.execution
xunit.runner.visualstudio
```

**Comment by loic-sharma** [comment]
Duplicate of #239, closing.

**Pull request closed without merging**